CanonicalLtd/documentation-builder
# Add notification support to markdown parser

This adds the styling and the markup generation to display [Vanilla notifications](https://docs.vanillaframework.io/en/patterns/notification) in documents.

Usage:

    !!! {type} "{title}":
        {content}

This will generate notification markup, e.g.:

    <div class="p-notification--{type}">
      <p class="p-notification__response">
	<span class="p-notification__status">{title}:</span>
	<span class="p-notification__line">{content}</span>
      </p>
    </div>

You can create notifications of types [supported by Vanilla framework](https://docs.vanillaframework.io/en/patterns/notification):

- Note
- Warning
- Positive
- Negative

Note, if `"{title}"` is ommitted (`!!! Note:`), the `type` will be used instead (e.g.: `<span>Note:</span>`). If it is left blank (`!!! Note "":`) then no title will be used.

Fixes #64.

QA
---

Test this out with maas-docs:

``` bash
git clone https://github.com/canonicalltd/maas-docs  # Get maas-docs
cd maas-docs  # go to maas-docs directory
git clone -b notifications https://github.com/nottrobin/documentation-builder  # clone down my branch
python3 -m venv env3  # Get ready to install in a local environment
env3/bin/pip install ./documentation-builder  # Install my branch
env3/bin/python documentation-builder/bin/documentation-builder --force  # Build the documentation
```

Now open `build/en/index.html` and see the notifications.

You can play around with the notifications by editing `en/index.md` and building the documentation again as mentioned above.
